CodeEdit/Features/SourceControl/Client/GitClient+Pull.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  GitClient+Pull.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 11/18/23.
+//
+
+import Foundation
+
+extension GitClient {
+    /// Pull changes from remote
+    func pullFromRemote(remote: String? = nil, branch: String? = nil, rebase: Bool? = nil) async throws {
```

Comment:
Why are the params optional, these would almost always be required.

Reply:
Users can do a `git pull` and it will pull their current branch without needing to specify. I am keeping close to how git works in this regard. Probably not extremely necessary.

Reply:
Using just git pull is not very effective when it comes to big projects. I would perhaps rethink how this is managed. Because if something goes wrong with git users won't be too happy